[Ezekiel Toh] iP

takufunkai opened the pull request:
# Duke Chatbot
- ~fast~ _VERY FAST_
- Efficient
- **Friendly!**

## How to start using
1. Download it from [here](https://nus-cs2103-ay2122s2.github.io/website/schedule/week4/project.html)
2. Install it
3. Start using!
...and it is **FREE!** 🤯

Features:
- [x] Task management
- [ ] Task auto-completion
- [ ] Remove task from real life (why do a task if you can get rid of it?)

> "This bot saved my life!" - Random User, 2019

---
If you are a Java developer, good. Otherwise, what are you waiting for? Learn some Java `c*de`!

```java
public class JavaSuperiority {
    public static void main(String[] args) {
        System.out.println("Java > all");
    }
}
```

Commit:
Add Gradle support

takufunkai committed:
Add greeting, exit and echo functionalities, and other text styles

takufunkai committed:
Add list functionality

takufunkai committed:
Rename userCommands to task

takufunkai committed:
Add usertask class and rename variables

takufunkai committed:
Change to switch statement

takufunkai committed:
Add mark and unmark functionalities

takufunkai committed:
Add todo class and new usertask package

takufunkai committed:
Add deadline class

takufunkai committed:
Add event class

takufunkai committed:
Add basic tests

takufunkai committed:
Add error handling

takufunkai committed:
Modify tests to verify exception handling

takufunkai committed:
Add delete functionality and relevant tests

takufunkai committed:
Remove unnecessary constant declarations

takufunkai committed:
Add enums

takufunkai committed:
Refactor switch cases into enums

takufunkai committed:
Fix bugs after refactor

takufunkai committed:
Add save mechanism

takufunkai committed:
Add restore mechanism

takufunkai committed:
Merge branch 'branch-Level-7'

takufunkai committed:
Merge branch branch-Level-8

takufunkai committed:
Refactor commands from enums to classes, and repackaged all classes

takufunkai committed:
Fix remaining bugs after refactoring

takufunkai committed:
Add basic junit tests

takufunkai committed:
Fix ui and style bugs

takufunkai committed:
Add metadata file for release version

takufunkai committed:
Add JavaDocs for all classes except usertask and utils pkgs

takufunkai committed:
Improve some styling to coding pattern guide

takufunkai committed:
Add find command

takufunkai committed:
Improve the flow of overriden methods in tasks

takufunkai committed:
Merge branch 'branch-A-JavaDoc'

takufunkai committed:
Merge branch 'branch-A-CodingStandard'

takufunkai committed:
Merge branch 'branch-Level-9'

takufunkai committed:
Merge branch 'add-gradle-support' into branch-A-Gradle

takufunkai committed:
Add checkstyle and suppressions xml files

tanyutao544 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,70 @@
+    /**
+     * Creates a new Duke chat-bot instance.
+     */
+    public Duke() {
```
Consider moving the constructor to the top

takufunkai replied:
Done, and along with all the Javadocs. Thank you for the kind words and feedback!

tanyutao544 commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,21 @@
+import duke.DukeException;
+import duke.textui.TextUi;
+import duke.usertask.TaskList;
+
```
Remember to add javadoc for new features

tanyutao544 commented on src/main/java/duke/usertask/Deadline.java:
```diff
@@ -0,0 +1,17 @@
+package duke.usertask;
+
```
Missing javadoc

tanyutao544 commented on src/main/java/duke/usertask/Event.java:
```diff
@@ -0,0 +1,17 @@
+package duke.usertask;
+
```
missing javadoc

tanyutao544 commented on src/main/java/duke/usertask/TaskList.java:
```diff
@@ -0,0 +1,92 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
```
missing javadoc

tanyutao544 commented on src/main/java/duke/usertask/ToDo.java:
```diff
@@ -0,0 +1,17 @@
+package duke.usertask;
+
```
missing javadoc

tanyutao544 commented on src/main/java/duke/usertask/UserTask.java:
```diff
@@ -0,0 +1,35 @@
+package duke.usertask;
+
+import java.util.Locale;
+
```
missing javadoc

tanyutao544 commented on src/main/java/duke/usertask/UserTaskWithTime.java:
```diff
@@ -0,0 +1,52 @@
+
+import static duke.utils.Utils.DATE_FORMAT;
+import static duke.utils.Utils.TIME_FORMAT;
+
```
missing javadoc

tanyutao544 commented on src/main/java/duke/utils/Utils.java:
```diff
@@ -0,0 +1,36 @@
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
```
missing javadoc

tanyutao544 reviewed (commented):
Ui print design is interesting, several classes missing javadoc, some methods can be abstracted. Consider adding more exception classes that extend from dukeException to reduce hard coding of error messages

takufunkai committed:
Fix non-javadoc related checkstyle errors

takufunkai committed:
Fix all checkstyle issues

takufunkai committed:
Fix deadline and event name bug

takufunkai committed:
Add all jfx tutorials

takufunkai committed:
Add css and other improvements to jfx tutorial end-product

takufunkai committed:
Add style and well-designed chat layout

takufunkai committed:
Add correct display pictures

takufunkai committed:
Migrate chat logic onto gui

takufunkai committed:
Add save handler

takufunkai committed:
Merge branch 'branch-Level-10'

takufunkai committed:
Remove text-based ui tests

takufunkai committed:
Add var-args

tsumian commented on src/main/java/duke/command/ListCommand.java:
```diff
@@ -0,0 +1,67 @@
+import duke.usertask.TaskList;
+
+/**
+ * ListCommand handles outputting the entire list of tasks that a user has at the current point in time.
+ * It has an optional argument field <code>dateFilter</code>, which will filter the list of tasks to output
+ * based on whether they 1. have a <code>dateTime</code> field, and 2. the date of the field is equal to the specified
+ * dateFilter supplied by the user. <b>Note:</b> time is irrelevant here, and filter works solely based on date.
+ */
```
I really like the level of detail you have for headers! It is consistent in almost every class I have seen!

takufunkai replied:
Thank you for your kind words and feedback!

tsumian commented on src/main/java/duke/gui/MainWindow.java:
```diff
@@ -0,0 +1,100 @@
+        dialogContainer.getChildren().add(DialogBox.getDukeDialog(out, dukeImage));
+    }
+
+    /**
+     * Creates two dialog boxes, one echoing user input and the other containing Duke's reply and then appends them to
+     * the dialog container. Clears the user input after processing.
+     */
```
Good that you included JavaDoc comments even for private methods. Makes it more informative.

tsumian commented on src/main/java/duke/storage/SaveHandler.java:
```diff
@@ -0,0 +1,128 @@
+    private static final String DATA_DIRECTORY = "data";
+    private static final String DATA_FILEPATH = DATA_DIRECTORY + "/duke.txt";
+
+    enum TaskCode {
+        T, D, E
+    }
```
Perhaps could be more consistent with Enum variable names.

tsumian commented on src/main/java/duke/usertask/UserTask.java:
```diff
@@ -0,0 +1,61 @@
+    /**
+     * Returns the string representation of the task in DukeSaveFormat.
+     *
+     * @return The string representation of the task in DUkeSaveFormat.
```
```suggestion
     * @return The string representation of the task in DukeSaveFormat.
```

tsumian replied:
```suggestion
     * @return The string representation of the task in DukeSaveFormat.
```

tsumian reviewed (commented):
Overall, your code is really good although it contains a lot of classes. JavaDocs are detailed enough and readable. Good job with the progress!

takufunkai committed:
Fix javafx version error

takufunkai committed:
Fix gradle build file not referencing launcher class

takufunkai committed:
Modify save behaviour from save on exit to immediate

Previously tasks were saved on graceful exit of the application. Now, all CRUD operations will reflect change in the data file immediately. Also added some missing JavaDocs.

takufunkai committed:
Improve readability and logical flow of LIST switch case

takufunkai committed:
Improve variable names and logical flow in Command.java

Variable names are ambiguous, and code is unreadable. Due
to having too many types of inputs that users
can supply, variables are carelessly named.

As more commands are being added, and new requirements
are being imposed on the commands, the code must be
easily maintainable by making it easy to understand
the current behaviour. This way, when adding new
requirements, bugs will not be accidentally introduced.

Let's
* rename the variables to be less ambiguous
* different case blocks to not share any variable names
if they are not declared before the switch case block.

If the variable names are being declared in an earlier
block, and being used in a later block, this means
that the scope of the variable can be further narrowed
down to be more specific. This increases readability
and reduces ambiguity.

takufunkai committed:
Merge branch 'master' into branch-A-CodeQuality

takufunkai committed:
Merge pull request #3 from takufunkai/branch-A-CodeQuality

Branch A-CodeQuality

takufunkai committed:
Add CI .yml file

takufunkai committed:
Merge branch 'branch-A-CI'

takufunkai committed:
Omit id in usertask tests

takufunkai committed:
Remove IO Redirection test from CI file

takufunkai committed:
Add client class

takufunkai committed:
Redesign command handling flow

takufunkai committed:
Add javadocs to new classes and refactor usertask method access modifiers

takufunkai committed:
Add ability to add clients

takufunkai committed:
Fix gender enum not catching correct value

takufunkai committed:
Rename command modules

takufunkai committed:
Add addClientForm component

takufunkai committed:
Improve layout of addClientForm

takufunkai committed:
Add addClient function in controllers

takufunkai committed:
Merge pull request #4 from takufunkai/branch-D-Clients

Add client functionality

takufunkai committed:
Add non-functional listview for clients

takufunkai committed:
Add client view and add functionality

takufunkai committed:
Fix savehandler overwriting data file

takufunkai committed:
Merge branch 'master' of https://github.com/takufunkai/ip

takufunkai committed:
Convert execute return to ResponseStatus

takufunkai committed:
Add missing javadocs

takufunkai committed:
Add background to scrollpane

takufunkai committed:
Rename SaveHandler to TasksStorage

takufunkai committed:
Add bye, todo command tests

takufunkai committed:
Add deadline and event tests

takufunkai committed:
Add listcommand test

takufunkai committed:
Add red background duke chat bubbles

takufunkai committed:
Fix checkstyle errors

takufunkai committed:
Fix style issues pointed by se bot

takufunkai committed:
Add product screenshot

takufunkai committed:
Update README

takufunkai committed:
Set theme jekyll-theme-tactile

takufunkai committed:
Fix log statements

takufunkai committed:
Remove duplicate Duke object

takufunkai committed:
Merge branch 'master' of https://github.com/takufunkai/ip
